### PR TITLE
GUI Notifications

### PR DESCRIPTION
**Pull request by jorenham**
Concerns #82 

Users can now get informed of errors, warnings and other information, through the GUI. Each notification has a display duration and after that expires, either the notification disappears or the next queued notification gets displayed.

_The code snippet below demonstrates the displaying of the exception thrown by the `openFiles` method:_

``` java
try {
    openFiles();
} catch (IOException e) {
    shout(NotificationController.NOTIFY, nf.getNotification(e));
}
```

### CODE REVIEW

**Comment by albert-smit** [comment]
looks good

File: src/main/java/nl/tudelft/lifetiles/notification/controller/NotificationController.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * A notification is displaying.
+     */
+    private boolean displaying;
+
+    @Override
+    public final void initialize(final URL location,
+            final ResourceBundle resources) {
+        final int initialCapacity = 10;
+        this.notifications = new PriorityQueue<>(initialCapacity,
+                (n1, n2) -> n1.getPriority() - n2.getPriority());
```
Comment (rutgervandenberg) [inline src/main/java/nl/tudelft/lifetiles/notification/controller/NotificationController.java]: Comparable needs to be consistent with equals. It should return 0 iff equals returns true (and hashmap returns the same value on both objects)

**Comment by rutgervandenberg** [comment]
- Notification.java: 
  - getDuration can be used without duration being initialized. Should initialize it in constructor.

**Comment by joswinter** [comment]
Looks good!

**Comment by alangerak** [comment]
Looks good

**Pull request merged**